0_RootFS/Rootfs/bundled/utils/bb.c

```diff
@@ -0,0 +1,54 @@
+#include <string.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <poll.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+// XXX: This is very bad style for a general server, but here
+// we know it will be used only for this limited purpose - do not use it
+// elsewhere.
```

Comment:
Can you elaborate on what exactly is bad about this code, and what should be improved in a different project, so that other people reading this can learn?

Reply:
The biggest thing is probably not buffering anything and just using raw syscalls everywhere. Also, the shutdown call is commented out because of that julia bug. Also big stack buffers are generally bad.